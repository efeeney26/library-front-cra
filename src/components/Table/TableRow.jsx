import React from 'react'

const TableRow = (props) => {
  return (
    <tr>
      <td>1</td>
      <td>2</td>
    </tr>
  )
}

export default TableRow
