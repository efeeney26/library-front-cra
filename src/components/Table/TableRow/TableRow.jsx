import React from 'react'
import PropTypes from 'prop-types'

import styles from './TableRow.module.css'

const TableRow = (props) => {
  const { data } = props
  return (
    data.map((dataItem) => (
      <tr key={dataItem}>
        {dataItem.map((valueItem) => (
          <td
            key={valueItem}
            className={styles.cell}
          >{valueItem}
          </td>
        ))}
      </tr>
    ))
  )
}

TableRow.propTypes = {
  data: PropTypes.arrayOf(PropTypes.array).isRequired
}

export default TableRow
