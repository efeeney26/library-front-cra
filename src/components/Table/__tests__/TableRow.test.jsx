import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import TableRow from '../TableRow/TableRow'

const props = {
  data: [
    ['testData1', 'testData2'],
    ['testData3', 'testData4']
  ]
}

test('TableRow snapshot', () => {
  const component = renderer.create(<TableRow {...props}/>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

describe('TableRow', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <TableRow
        {...props}
      />
    )
  })

  it('TableHead', () => {
    expect(wrapper.find('td').length).toBe(4)
  })
})
