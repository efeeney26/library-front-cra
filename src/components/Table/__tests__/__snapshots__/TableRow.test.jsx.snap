// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow snapshot 1`] = `
Array [
  <tr>
    <td
      className="cell"
    >
      testData1
    </td>
    <td
      className="cell"
    >
      testData2
    </td>
  </tr>,
  <tr>
    <td
      className="cell"
    >
      testData3
    </td>
    <td
      className="cell"
    >
      testData4
    </td>
  </tr>,
]
`;
